istioctl/cmd/istioctl/inject.go

```diff
@@ -353,6 +371,12 @@ func init() {
 			"are excluded.")
 	injectCmd.PersistentFlags().BoolVar(&debugMode, "debug", false, "Use debug images and settings for the sidecar")
 
+	deprecateFlags := []string{"coreDump", "imagePullPolicy", "includeIPRanges", "excludeIPRanges",
```

Comment:
The `hub` and `tag` flags and `ISTIO_PROXY_IMAGE` envvar should also be deprecated.